src/lib/twitter/notifyTwitter.ts

```diff
@@ -0,0 +1,23 @@
+import TwitterAPI from 'twitter-api-v2';
+import { NFTSale } from "lib/marketplaces";
+import axios from 'axios';
+
+export default async function notifyTwitter(twitterClient: TwitterAPI, nftSale: NFTSale) {
+    const hasImage = Boolean(nftSale.nftData?.image);
+    const data = hasImage ? await getImageDataFromUrl(nftSale.nftData?.image as string) : null;
```

Comment:
Looks like data is only used for the upload before. I think it's easier to read of the code stays closer to here
```ts
const mediaArr: string[] = [];
if (hasImage) {
  const mediaArr: string[] = [];
  const data = await getImageDataFromUrl(nftSale.nftData?.image as string);
  const media = await twitterClient.v1.uploadMedia(data as Buffer, { type: 'gif' });
  mediaArr.push(media);
}
```

Reply:
Done!